File: Code/branches/PHYSOR/cpp/test/test_BankSource.cpp
/*test_BankSource.cpp
 $Author: jlconlin $
 $Id: test_BankSource.cpp 236 2008-01-03 21:14:32Z jlconlin $
 $Revision: 236 $
 $Date: 2008-01-03 14:14:32 -0700 (Thu, 03 Jan 2008) $

 This file is used for testing the BankSource class.

 */
#include<iostream>
#include<vector>

#include "RandomLib/Random.hpp"
#include "Field.hh"
#include "BankSource.h"
#include "Particle.h"
#include "Material.h"

using std::vector;
using std::cout;

int main(){
    vector<unsigned long> seed(2);
    seed[0] = RandomLib::Random::SeedWord();
    cout << "Master seed: " << seed[0] << "\n";
    
    unsigned int N  = 100000;
    unsigned int numZones = 10;
    
    cout << "Here I create a mesh of " << numZones 
              << " zones each of width 1.0.\n";
    Mesh meshU(1.0, 1.0, numZones);
    
    cout << "Here I am creating a Field of identical Materials.\n";
    vector<Material> myMaterials(numZones);
    for( vector<Material>::iterator matIter = myMaterials.begin();
        matIter != myMaterials.end(); ++matIter ){
        
        *matIter = Material(0.3, 0.3, 0.4, 1.0);
    }
    Field<Zone, Material> geometry(meshU, myMaterials);

    
    //Uniform source
    cout << "Here I define/create the source probabilities for "
              << "each Zone.\n";
    vector<double> sourceProbability(numZones, 1.0);
    Field<Zone, double> uSource(meshU, sourceProbability);
    
    cout << "BankSource will sample " << N << " particles.\n";
    BankSource* uBank = new BankSource(uSource, meshU, seed, N);
    cout << "Size of BankSource: " << uBank->size() << "\n";
    cout << "\nDiscretizing BankSource...\n";
    cout << "\tNumber of Particles in each Zone: \n";
    
    vector<double> disc = uBank->discretized(meshU);
    for( vector<double>::iterator dIter = disc.begin();
        dIter != disc.end(); ++dIter ){
        
        cout << *dIter << "\n";
    }
    delete uBank;
    return 0;
}

File: Code/branches/Pre-Prospectus/cpp/FissionSource/test_BankSource.py
__id__       = "$Id: test_BankSource.py 230 2008-01-02 15:12:50Z jlconlin $"
__author__   = "$Author: jlconlin $"
__version__ = " $Revision: 230 $"
__date__     = "$Date: 2008-01-02 08:12:50 -0700 (Wed, 02 Jan 2008) $"

"""
test_BankSource.py is designed to test the BankSource class.
"""

from Transport.Geometry import Field, Mesh
from Transport import Material
from Transport.FissionSource import BankSource

N = 100000   # Number of particles

numZones = 10
ZoneLengths = [1.0]*10
print "Here I create a Mesh of %s zones each of length 1.0" %(numZones)
myMesh = Mesh.Mesh(ZoneLengths)  # Mesh with specified non-uniform ZoneLengths

print "Here I create a Field of the same size as the Mesh."
print "And I fill it with the same material in each Zone."
Mat1 = Material.Material(0.5, 0.5, 0.0, 1.0)  # 'Standard' material
print "The Material I use is: %s" %(Mat1)
myMaterial = [Mat1 for i in xrange(numZones)]
vecMats = Material.vectormats(myMaterial)

print "Now I define source probabilities for each Zone."
vecProbabilities = Field.DVector([1.0 for i in xrange(numZones)])

F = Field.FieldZoneMat(myMesh, vecMats)
ProbField = Field.FieldZoneDouble(myMesh, vecProbabilities)

print "Creating (not so) random seed..."
seed = BankSource.ULongVector([1])

print "Now BankSource will sample %i particles" %(N)
uniBank = BankSource.BankSource(ProbField, myMesh, seed, N)

print "The discretized source is: ",
disc = uniBank.discretized(myMesh)
print "%s" %(disc)
